Spec/RealtimeClientPresence.swift

```diff
@@ -957,6 +957,94 @@ class RealtimeClientPresence: QuickSpec {
                     expect(compareForNewnessMethodCalls) == 1
                 }
 
+                // RTP2b
+                context("compare for newness") {
+
+                    // RTP2b1
+                    it("presence message has a connectionId which is not an initial substring of its id") {
+                        let options = AblyTests.commonAppSetup()
+
+                        let clientSubscribed = ARTRealtime(options: options)
+                        defer { clientSubscribed.dispose(); clientSubscribed.close() }
+                        let channelSubscribed = clientSubscribed.channels.get("foo")
+                        channelSubscribed.attach()
+
+                        let clientPresentMember = ARTRealtime(options: options)
+                        defer { clientPresentMember.dispose(); clientPresentMember.close() }
+                        let channelPresentMember = clientPresentMember.channels.get("foo")
+
+                        var hasInconsistentConnectionIdMethodCalls = 0
+                        let hook = channelSubscribed.presenceMap.testSuite_injectIntoMethodAfter(NSSelectorFromString("hasInconsistentConnectionId")) {
```

Comment:
This type of test condition, IMHO, is bad.  You're testing that this test passes because a private method is being called, instead of simply testing that the behaviour and outcome is correct.  I would prefer that you don't check these types of things unless there is no other way (i.e. you want to confirm that an internal event has happened perhaps).  In this case however, there are plenty of ways to confirm the intended functionality - you could enter a few members in the client, then inject synthesized `ProtocolMessage` events from Ably so that you get a realistic set of protocol messages sent to the client which should prove that compare by timestamp is working i.e. some are discarded, and some are applied.  I implemented all of this yesterday in fact, see https://github.com/ably/ably-ruby/pull/99/commits/e699106b0c55a82040336b4aa3a58a95f620546f#diff-c9dcfc1419f386559fc4a8a0ef368c99R597